packages/shared/src/types/addresses.ts

```diff
@@ -20,10 +21,33 @@ export type AddressSettings = {
 
 export type AddressIndex = number
 
+// index: number
+// isDefault: boolean
+// color: string
+// label?: string
 export type AddressMetadata = AddressSettings & {
   index: AddressIndex
 }
 
+// hash: AddressHash
+// index: number
+// isDefault: boolean
+// color: string
+// label?: string
+export type AddressBase = AddressMetadata & {
+  hash: AddressHash
+}
+
+// hash: AddressHash
+// index: number
+// group: number
+// isDefault: boolean
+// color: string
+// label?: string
+export type Address = AddressBase & {
```

Comment:
Does it make sense to have both `AddressBase` and `Address` when only one field is different? 
Would it make sense to simply have: 

```ts 
type Address = AddressMetadata & { 
  hash: AddressHash
  group: number
}
```

Reply:
Yes we'd need to derive the group by default, but it would prevent confusing types.